src/QsCompiler/Core/TransformationOptions.fs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.Core
+
+
+/// Used to configure the behavior of the default implementations for transformations.
+type TransformationOptions = {
+    /// Disables the transformation at the transformation root, 
+    /// meaning the transformation won't recur into leaf nodes or subnodes. 
+    Disable : bool
```

Comment:
I am not sure why this option is needed. Can you explain in what scenarios it would be useful for?

Reply:
Each subtransformation can be configured upon construction. You would set these to disabled e.g. if you don't want to do expression transformations. This is replacing the bool that the old setup took as constructor argument. Does this clarify things?